### DISCUSSION TITLE
How can I call `auth.uid()` and other functions within queries from the supabase client?

### DISCUSSION THREAD

**Discussion by park-jemin** [supabase/supabase#26786] [Questions]
For example, with the JS client, I would need to call `supabase.auth.getUser()` to get the current user and id. However, that adds a bit of overhead, and I would like to call the function directly in a query, e.g.,

```ts
const result = await supabase
      .from('users')
      .select(`*`)
      .eq('id', 'auth.uid()') // Something like this
```

How could I go about achieving this without using an RPC?

**Comment by GaryAustin1** [comment]
You can't.  If you have an RLS policy though you could enforce that with RLS.  So you could use the "unsecure" session.user.id in the call and trust that RLS will protect the table access by checking id=auth.uid() in the RLS.
